File: S_3/Programação Orientada a Objetos/Exercicio2.java
import java.io.IOException;
import java.lang.Math;

public class Exercicio2 {

    public static double lerNum() {
        double x = 0;
        boolean talkey = false;
        while (!talkey) {
            try {
                x = EntradaTeclado.leDouble();
                return x;
            } catch (Exception e) {
                System.out.println("Número incorreto! Digite novamente:");
            }
        }
        return x;
    }

    public static void bhaskara(double a, double b, double c) {
        double delta = b * b - 4 * a * c;
        if (delta >= 0) {
            double s1 = (-b + Math.sqrt(delta)) / 2 * a;
            double s2 = (-b - Math.sqrt(delta)) / 2 * a;
            System.out.println("O conjunto solução da equação é : S = [" + s1 + "," + s2 + "]");
        } else {
            System.out.println("Não possui raízes reais!");
        }
    }

    public static void main(String[] args) throws IOException {
        double c1 = 0, c2 = 0, c3 = 0;
        System.out.println("Digite o primeiro coeficiente:");
        c1 = lerNum();
        System.out.println("Digite o segunda coeficiente:");
        c2 = lerNum();
        System.out.println("Digite o terceiro coeficiente:");
        c3 = lerNum();
        bhaskara(c1,c2,c3);
    }
}

File: S_3/Programação Orientada a Objetos/Exercicio2.py
import math
def bhaskara(a, b, c):
    delta = b*b - 4 * a * c
    if delta >= 0:
        x1 = (-b + math.pow(delta,0.5))/(2*a)
        x2 = (-b - math.pow(delta,0.5))/(2*a)
        print("Tem duas raizes "+str(x1)+" e "+str(x2))
    else:
        print("Não tem raizes")
    return
def lerNum():
    while True:
        try:
            x = float(input())
            break
        except ValueError:
            print("Não é número, digite novamente:")
    return x
print("Digite o primeiro coeficiente:")
coef_a = lerNum()
print("Digite o segundo coeficiente:")
coef_b = lerNum()
print("Digite o terceiro coeficiente:")
coef_c = lerNum()
bhaskara(coef_a,coef_b,coef_c)
